### PR TITLE
nixos/dokuwiki: Finally remove extraConfig

### PR DESCRIPTION
**Pull request by e1mo**
###### Description of changes

After the introduction of structured settings in #208299 the old string-style options  / types which were kept for compatibility are now removed in preparation for the 23.05 release.

I went to the extra effort for both the renamed and removed options to show the full option paths to offer users upgrading from 22.11 to 23.05 actually useful warnings that make migrating as effortless as possible.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/dokuwiki.nix [hunk 1]
```diff
@@ -112,38 +113,67 @@ let
...
           See <https://www.dokuwiki.org/acl#background_info> for explanation
         '';
         example = "read";
       };
-
     };
   };
 
-  siteOpts = { config, lib, name, ... }:
+  # Lite re-implementation of mkRenamedOptionModule and doRename so that we can show the full path
```
Comment (K900) [inline nixos/modules/services/web-apps/dokuwiki.nix:125]: Any reason to _not_ merge this into the primary implementation of `mkRenamedOptionModule`?

Comment (AndersonTorres) [reply nixos/modules/services/web-apps/dokuwiki.nix:125]: It should be more explicit on the comment. Something like "the functionality X is not provided by the current implementation of Y".

File: nixos/modules/services/web-apps/dokuwiki.nix [hunk 2]
```diff
@@ -112,38 +113,67 @@ let
...
-              Please pass structured settings instead.
-            '')
-            ++ (optional (config.extraConfig != null) ''
-              services.dokuwiki.sites.${name}.extraConfig is deprecated and will be removed in the future.
-              Please pass structured settings to services.dokuwiki.sites.${name}.settings instead.
-            '')
-          ;
+        (mkRenamed [ "aclUse" ] [ "settings" "useacl" ])
+        (mkRenamed [ "superUser" ] [ "settings" "superuser" ])
+        (mkRenamed [ "disableActions" ] [ "settings"  "disableactions" ])
```
Comment (AndersonTorres) [inline nixos/modules/services/web-apps/dokuwiki.nix:152]: Is there any reason to not keep the `camelCase`?

File: nixos/doc/manual/release-notes/rl-2305.section.md [hunk 3]
```diff
@@ -279,9 +279,9 @@ In addition to numerous new and upgraded packages, this release has the followin
   `services.dnsmasq.extraConfig` will be deprecated when NixOS 22.11 reaches
   end of life.
 
-- The `dokuwiki` service now takes configuration via the `services.dokuwiki.sites.<name>.settings` attribute set, `extraConfig` is deprecated and will be removed.
-  The `{aclUse,superUser,disableActions}` attributes have been renamed, `pluginsConfig` now also accepts an attribute set of booleans, passing plain PHP is deprecated.
-  Same applies to `acl` which now also accepts structured settings.
+- The `dokuwiki` service now takes configuration via the `services.dokuwiki.sites.<name>.settings` attribute set, `extraConfig` has been removed.
+  The `{aclUse,superUser,disableActions}` attributes have been renamed accordingly, `pluginsConfig` now only accepts an attribute set of booleans, passing plain PHP is no longer possible.
+  Same applies to `acl` which now also only accepts structured settings.
```
Comment (AndersonTorres) [inline nixos/doc/manual/release-notes/rl-2305.section.md:284]:
```suggestion
- The `dokuwiki` service is now configured via `services.dokuwiki.sites.<name>.settings` attribute set; `extraConfig` has been removed.
  The `{aclUse,superUser,disableActions}` attributes have been renamed accordingly. `pluginsConfig` now only accepts an attribute set of booleans.
Passing plain PHP is no longer possible.
  Same applies to `acl` which now also only accepts structured `settings`.
```

File: nixos/modules/services/web-apps/dokuwiki.nix (hunk 2 above)
Comment (e1mo) [reply nixos/modules/services/web-apps/dokuwiki.nix:154]: These options were removed/renamed in #208299 with the migration to only use structured settings (in line with [RFC42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md)). I kept them as a rename for backwards-compatibility (only updated it with the new function for showing the full path) with the intent of leaving it there until at least 23.11 to not break anything we don't need to.

**Review by AndersonTorres — APPROVED** [review]

**Comment by onny** [comment]
Works fine for me, thank you!

**Pull request merged**